From 616f135a309141a7373094275b4af811f9c51d13 Mon Sep 17 00:00:00 2001
From: Rebecca Chang Swee Fun <rebecca.chang@starfivetech.com>
Date: Fri, 17 Jun 2022 00:29:10 +0000
Subject: [PATCH 02/51] chrome: installer: add riscv64 as one of the target
 arch

Signed-off-by: Rebecca Chang Swee Fun <rebecca.chang@starfivetech.com>
---
 chrome/installer/linux/BUILD.gn | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/chrome/installer/linux/BUILD.gn b/chrome/installer/linux/BUILD.gn
index 6623fff2ba8a..104b8b5bb135 100644
--- a/chrome/installer/linux/BUILD.gn
+++ b/chrome/installer/linux/BUILD.gn
@@ -477,6 +477,8 @@ template("linux_package") {
       deb_arch = "mipsel"
     } else if (current_cpu == "mips64el") {
       deb_arch = "mips64el"
+    } else if (current_cpu == "riscv64") {
+      deb_arch = "riscv64"
     } else {
       assert(false, "Linux installer not configured for this architecture.")
     }
@@ -526,6 +528,8 @@ template("linux_package") {
         rpm_arch = "mipsel"
       } else if (current_cpu == "mips64el") {
         rpm_arch = "mips64el"
+      } else if (current_cpu == "riscv64") {
+        rpm_arch = "riscv64"
       } else {
         assert(false, "Linux installer not configured for this architecture.")
       }
@@ -579,6 +583,8 @@ template("linux_package") {
           snap_arch = "armhf"
         } else if (current_cpu == "arm64") {
           snap_arch = "arm64"
+        } else if (current_cpu == "riscv64") {
+          snap_arch = "riscv64"
         } else {
           assert(false, "Linux installer not configured for this architecture.")
         }
-- 
2.25.1

