From 5926c534338961fd9b431ca00bba3b0abd4f85ba Mon Sep 17 00:00:00 2001
From: Rebecca Chang Swee Fun <rebecca.chang@starfivetech.com>
Date: Wed, 30 Mar 2022 05:44:17 +0000
Subject: [PATCH 05/18] base: process: add riscv64 arch definition

Signed-off-by: Rebecca Chang Swee Fun <rebecca.chang@starfivetech.com>
---
 base/process/launch_posix.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/base/process/launch_posix.cc b/base/process/launch_posix.cc
index 5130a0baa2c7d..add3ca595da80 100644
--- a/base/process/launch_posix.cc
+++ b/base/process/launch_posix.cc
@@ -707,7 +707,7 @@ NOINLINE pid_t CloneAndLongjmpInChild(unsigned long flags,
   alignas(16) char stack_buf[PTHREAD_STACK_MIN];
 #if defined(ARCH_CPU_X86_FAMILY) || defined(ARCH_CPU_ARM_FAMILY) ||   \
     defined(ARCH_CPU_MIPS_FAMILY) || defined(ARCH_CPU_S390_FAMILY) || \
-    defined(ARCH_CPU_PPC64_FAMILY)
+    defined(ARCH_CPU_PPC64_FAMILY) || defined(ARCH_CPU_RISCV_FAMILY)
   // The stack grows downward.
   void* stack = stack_buf + sizeof(stack_buf);
 #else
-- 
2.25.1

