From f51a84eac8c9adaa8be60b93f457e3a128a43b46 Mon Sep 17 00:00:00 2001
From: Rebecca Chang Swee Fun <rebecca.chang@starfivetech.com>
Date: Tue, 6 Sep 2022 02:29:45 -0400
Subject: [PATCH 09/68] DEPS: use custom dawn repo

Signed-off-by: Rebecca Chang Swee Fun <rebecca.chang@starfivetech.com>
---
 DEPS | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/DEPS b/DEPS
index f90dee1c598b..8b9d38dbc05c 100644
--- a/DEPS
+++ b/DEPS
@@ -366,7 +366,7 @@ vars = {
   # Three lines of non-changing comments so that
   # the commit queue can handle CLs rolling feed
   # and whatever else without interference from each other.
-  'dawn_revision': 'fa8cc68ff7c055512e83a538e5517400f5f053bc',
+  'dawn_revision': '864f58d98e0fdcc4a3203bb4cf4cefdc67307c41',
   # Three lines of non-changing comments so that
   # the commit queue can handle CLs rolling feed
   # and whatever else without interference from each other.
@@ -1011,7 +1011,7 @@ deps = {
     Var('chromium_git') + '/external/github.com/videolan/dav1d.git' + '@' + '87f9a81cd770e49394a45deca7a3df41243de00b',
 
   'src/third_party/dawn':
-    Var('dawn_git') + '/dawn.git' + '@' +  Var('dawn_revision'),
+    Var('reb_git') + '/dawn.git' + '@' +  Var('dawn_revision'),
 
   'src/third_party/libjxl/src':
     Var('chromium_git') + '/external/gitlab.com/wg1/jpeg-xl.git' + '@' + Var('libjxl_revision'),
-- 
2.30.2

