From 0f699d5bcbfb5f97f2f78054bd9c0f8ac913f328 Mon Sep 17 00:00:00 2001
From: Rebecca Chang Swee Fun <rebecca.chang@starfivetech.com>
Date: Fri, 17 Jun 2022 00:29:10 +0000
Subject: [PATCH 11/68] chrome: installer: add riscv64 as one of the target
 arch

Signed-off-by: Rebecca Chang Swee Fun <rebecca.chang@starfivetech.com>
---
 chrome/installer/linux/BUILD.gn | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/chrome/installer/linux/BUILD.gn b/chrome/installer/linux/BUILD.gn
index 820ea5f4b1ab..56dcf19a2cbd 100644
--- a/chrome/installer/linux/BUILD.gn
+++ b/chrome/installer/linux/BUILD.gn
@@ -462,6 +462,8 @@ template("linux_package") {
       deb_arch = "mipsel"
     } else if (current_cpu == "mips64el") {
       deb_arch = "mips64el"
+    } else if (current_cpu == "riscv64") {
+      deb_arch = "riscv64"
     } else {
       assert(false, "Linux installer not configured for this architecture.")
     }
@@ -511,6 +513,8 @@ template("linux_package") {
         rpm_arch = "mipsel"
       } else if (current_cpu == "mips64el") {
         rpm_arch = "mips64el"
+      } else if (current_cpu == "riscv64") {
+        rpm_arch = "riscv64"
       } else {
         assert(false, "Linux installer not configured for this architecture.")
       }
@@ -564,6 +568,8 @@ template("linux_package") {
           snap_arch = "armhf"
         } else if (current_cpu == "arm64") {
           snap_arch = "arm64"
+        } else if (current_cpu == "riscv64") {
+          snap_arch = "riscv64"
         } else {
           assert(false, "Linux installer not configured for this architecture.")
         }
-- 
2.30.2

