From 5ea13676fb160bb724dacde6730e109a71192061 Mon Sep 17 00:00:00 2001
From: Rebecca Chang Swee Fun <rebecca.chang@starfivetech.com>
Date: Mon, 1 Aug 2022 17:27:10 +0800
Subject: [PATCH 15/68] tflite_support: add cflags to resolve build issues

Signed-off-by: Rebecca Chang Swee Fun <rebecca.chang@starfivetech.com>
---
 third_party/tflite_support/BUILD.gn | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/third_party/tflite_support/BUILD.gn b/third_party/tflite_support/BUILD.gn
index 6d9c6a09d9f2..284ef4c742d2 100644
--- a/third_party/tflite_support/BUILD.gn
+++ b/third_party/tflite_support/BUILD.gn
@@ -40,6 +40,8 @@ config("tflite_support_flags") {
     "-Wno-ignored-attributes",
     "-Wno-deprecated-declarations",
     "-Wno-unused-variable",
+    "-Wno-defaulted-function-deleted",
+    "-Wno-absolute-value",
   ]
   if (!is_win) {
     cflags_cc = [ "-frtti" ]
-- 
2.30.2

