From 63c08e63eb74423aeea48903105cce7bc33a49a4 Mon Sep 17 00:00:00 2001
From: Rebecca Chang Swee Fun <rebecca.chang@starfivetech.com>
Date: Thu, 23 Jun 2022 15:33:33 +0000
Subject: [PATCH 16/51] DEPS: bump swiftshader to latest

Signed-off-by: Rebecca Chang Swee Fun <rebecca.chang@starfivetech.com>
---
 DEPS | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/DEPS b/DEPS
index a1a0c1061451..338adadd4456 100644
--- a/DEPS
+++ b/DEPS
@@ -266,7 +266,7 @@ vars = {
   # Three lines of non-changing comments so that
   # the commit queue can handle CLs rolling SwiftShader
   # and whatever else without interference from each other.
-  'swiftshader_revision': 'd01dca1d18d03f055d0dabd99a210b5f666715ed',
+  'swiftshader_revision': '1b159f6cdecba5e29858e50a998951d6c5ef91d3',
   # Three lines of non-changing comments so that
   # the commit queue can handle CLs rolling PDFium
   # and whatever else without interference from each other.
-- 
2.25.1

