From ae85e4121756a0e23ae01eabe1d45afc96d1645c Mon Sep 17 00:00:00 2001
From: Rebecca Chang Swee Fun <rebecca.chang@starfivetech.com>
Date: Mon, 25 Jul 2022 14:45:42 +0800
Subject: [PATCH 21/51] DEPS: use DEPS to pull in v8 source tree

Signed-off-by: Rebecca Chang Swee Fun <rebecca.chang@starfivetech.com>
---
 DEPS | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/DEPS b/DEPS
index 0be6053c3f4b..ec62ba17f53f 100644
--- a/DEPS
+++ b/DEPS
@@ -259,7 +259,7 @@ vars = {
   # Three lines of non-changing comments so that
   # the commit queue can handle CLs rolling V8
   # and whatever else without interference from each other.
-  'v8_revision': 'e5f13e0cd10e245d822f856d17b347c77ccc7593',
+  'v8_revision': 'e104a918db1ae901f9477a57fa179c26778e639a',
   # Three lines of non-changing comments so that
   # the commit queue can handle CLs rolling ANGLE
   # and whatever else without interference from each other.
@@ -1826,8 +1826,8 @@ deps = {
       'condition': 'checkout_mac',
   },
 
-  #'src/v8':
-  #  Var('chromium_git') + '/v8/v8.git' + '@' +  Var('v8_revision'),
+  'src/v8':
+    Var('reb_git') + '/v8.git' + '@' +  Var('v8_revision'),
 
   'src-internal': {
     'url': 'https://chrome-internal.googlesource.com/chrome/src-internal.git@a911c3b7bc546c3f876a2fe98617f0509e5fc67f',
-- 
2.25.1

