From 5af6df583558e7f7eb671f6a8caee64324cf6067 Mon Sep 17 00:00:00 2001
From: Rebecca Chang Swee Fun <rebecca.chang@starfivetech.com>
Date: Wed, 19 Jan 2022 05:48:04 +0000
Subject: [PATCH 22/23] build: gn_run_binary.py: do not run binary from ./

Signed-off-by: Rebecca Chang Swee Fun <rebecca.chang@starfivetech.com>
---
 build/gn_run_binary.py | 2 --
 1 file changed, 2 deletions(-)

diff --git a/build/gn_run_binary.py b/build/gn_run_binary.py
index d7e2926fa4fba..f071a78d0fd83 100644
--- a/build/gn_run_binary.py
+++ b/build/gn_run_binary.py
@@ -18,8 +18,6 @@ import sys
 # may prefix it with "./" to avoid picking up system versions that might
 # also be on the path.
 path = sys.argv[1]
-if not os.path.isabs(path):
-  path = './' + path
 
 # The rest of the arguments are passed directly to the executable.
 args = [path] + sys.argv[2:]
-- 
2.25.1

