From 695c3cbdc74c0a661bbbfaa7894dd0a35b2de081 Mon Sep 17 00:00:00 2001
From: Rebecca Chang Swee Fun <rebecca.chang@starfivetech.com>
Date: Tue, 15 Feb 2022 02:37:36 +0000
Subject: [PATCH 23/23] build: config: compiler: add shared object path to
 compiler

Signed-off-by: Rebecca Chang Swee Fun <rebecca.chang@starfivetech.com>
---
 build/config/compiler/BUILD.gn | 1 +
 1 file changed, 1 insertion(+)

diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index 9aa96a006cfde..2f0a0177f5bb0 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -444,6 +444,7 @@ config("compiler") {
     ldflags += [
       "-Wl,-z,noexecstack",
       "-Wl,-z,relro",
+      "-Wl,-L../../build/linux/debian_sid_riscv64-sysroot/usr/lib/riscv64-linux-gnu",
     ]
 
     if (!is_component_build) {
-- 
2.25.1

