From 7c9704325f382d401dc5cb5e3f4ad53a3875a926 Mon Sep 17 00:00:00 2001
From: Rebecca Chang Swee Fun <rebecca.chang@starfivetech.com>
Date: Fri, 29 Jul 2022 14:50:36 +0800
Subject: [PATCH 23/51] DEPS: update commit hash for ffmpeg

Signed-off-by: Rebecca Chang Swee Fun <rebecca.chang@starfivetech.com>
---
 DEPS | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/DEPS b/DEPS
index add99378747a..0a16f35dd321 100644
--- a/DEPS
+++ b/DEPS
@@ -1187,7 +1187,7 @@ deps = {
     Var('chromium_git') + '/external/github.com/google/farmhash.git' + '@' + '816a4ae622e964763ca0862d9dbd19324a1eaf45',
 
   'src/third_party/ffmpeg':
-    Var('reb_git') + '/chromium-ffmpeg.git' + '@' + '1a69da082435b0e7b56f5c09e4621c29aae24077',
+    Var('reb_git') + '/chromium-ffmpeg.git' + '@' + '63c83e8f91ece09847337856ad7b14a1c25d3c77',
 
   'src/third_party/flac':
     Var('chromium_git') + '/chromium/deps/flac.git' + '@' + 'af862024c8c8fa0ae07ced05e89013d881b00596',
-- 
2.25.1

