From aead4720afd4a774cb14520cfb8b581ee78a9c8b Mon Sep 17 00:00:00 2001
From: Rebecca Chang Swee Fun <rebecca.chang@starfivetech.com>
Date: Wed, 22 Jun 2022 08:06:36 +0000
Subject: [PATCH 27/51] build: toolchain: linux: enable clang toolchain for
 riscv64

Signed-off-by: Rebecca Chang Swee Fun <rebecca.chang@starfivetech.com>
---
 build/toolchain/linux/BUILD.gn | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/build/toolchain/linux/BUILD.gn b/build/toolchain/linux/BUILD.gn
index bc17e80e6135..64face8eff5c 100644
--- a/build/toolchain/linux/BUILD.gn
+++ b/build/toolchain/linux/BUILD.gn
@@ -290,6 +290,15 @@ gcc_toolchain("mips64el") {
   }
 }
 
+clang_toolchain("clang_riscv64") {
+  enable_linker_map = true
+  toolchain_args = {
+    current_cpu = "riscv64"
+    current_os = "linux"
+    is_clang = true
+  }
+}
+
 gcc_toolchain("riscv64") {
   toolprefix = "riscv64-linux-gnu"
 
-- 
2.25.1

