From a318817c2321b2e0c8a99d519e1fdcbab4f73585 Mon Sep 17 00:00:00 2001
From: Tien Hock Loh <tienhock.loh@starfivetech.com>
Date: Thu, 11 Aug 2022 08:05:06 +0000
Subject: [PATCH 30/51] build: config: add v8 target cpu for riscv64

Signed-off-by: Tien Hock Loh <tienhock.loh@starfivetech.com>
---
 build/config/v8_target_cpu.gni | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/build/config/v8_target_cpu.gni b/build/config/v8_target_cpu.gni
index 305981f3fa53..b5767be5bc93 100644
--- a/build/config/v8_target_cpu.gni
+++ b/build/config/v8_target_cpu.gni
@@ -36,6 +36,8 @@ declare_args() {
 if (v8_target_cpu == "") {
   if (current_toolchain == "//build/toolchain/linux:clang_x64_v8_arm64") {
     v8_target_cpu = "arm64"
+  } else if (current_toolchain == "//build/toolchain/linux:clang_x64_v8_riscv64") {
+    v8_target_cpu = "riscv64"
   } else if (current_toolchain == "//build/toolchain/linux:clang_x86_v8_arm") {
     v8_target_cpu = "arm"
   } else if (current_toolchain ==
-- 
2.25.1

