From fb7083aa25aef30842026338e8ba5aaf6a208d66 Mon Sep 17 00:00:00 2001
From: Rebecca Chang Swee Fun <rebecca.chang@starfivetech.com>
Date: Mon, 4 Jul 2022 08:57:14 +0000
Subject: [PATCH 30/68] build: config: BUILD.gn: add atomic build flag

Signed-off-by: Rebecca Chang Swee Fun <rebecca.chang@starfivetech.com>
---
 build/config/BUILD.gn | 1 +
 1 file changed, 1 insertion(+)

diff --git a/build/config/BUILD.gn b/build/config/BUILD.gn
index ac726a32b264..f2aa6faf6369 100644
--- a/build/config/BUILD.gn
+++ b/build/config/BUILD.gn
@@ -227,6 +227,7 @@ config("default_libs") {
       "dl",
       "pthread",
       "rt",
+      "atomic",
     ]
   }
 }
-- 
2.30.2

