From c7e20ee170c73c4ab553c4da2d7191aa5b99e1ab Mon Sep 17 00:00:00 2001
From: Tien Hock Loh <tienhock.loh@starfivetech.com>
Date: Thu, 11 Aug 2022 08:06:56 +0000
Subject: [PATCH 31/51] build: toolchain: linux: do not use lld by default

Signed-off-by: Tien Hock Loh <tienhock.loh@starfivetech.com>
---
 build/toolchain/linux/BUILD.gn | 1 +
 1 file changed, 1 insertion(+)

diff --git a/build/toolchain/linux/BUILD.gn b/build/toolchain/linux/BUILD.gn
index 64face8eff5c..b32804c1752d 100644
--- a/build/toolchain/linux/BUILD.gn
+++ b/build/toolchain/linux/BUILD.gn
@@ -168,6 +168,7 @@ clang_v8_toolchain("clang_x64_v8_riscv64") {
   toolchain_args = {
     current_cpu = "x64"
     v8_current_cpu = "riscv64"
+    use_lld = false
   }
 }
 
-- 
2.25.1

