From b09d6492e224bfbecebe414b2c53baba11930a51 Mon Sep 17 00:00:00 2001
From: Tien Hock Loh <tienhock.loh@starfivetech.com>
Date: Fri, 5 Aug 2022 03:06:18 -0400
Subject: [PATCH 33/51] build: config: compiler: skip warning for ffmpeg

Signed-off-by: Tien Hock Loh <tienhock.loh@starfivetech.com>
---
 build/config/compiler/BUILD.gn | 1 +
 1 file changed, 1 insertion(+)

diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index 2ba6030cc544..c2fce9eb8ede 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -1936,6 +1936,7 @@ config("wglobal_constructors") {
 config("wexit_time_destructors") {
   if (is_clang) {
     cflags = [ "-Wexit-time-destructors" ]
+    cflags += [ "-Wno-deprecated-declarations" ]
   }
 }
 
-- 
2.25.1

