From cb072c1ca121c85743f0623e7d4f30cd02d2663c Mon Sep 17 00:00:00 2001
From: Rebecca Chang Swee Fun <rebecca.chang@starfivetech.com>
Date: Wed, 22 Jun 2022 08:05:49 +0000
Subject: [PATCH 38/51] add path to riscv64 sysroot

Signed-off-by: Rebecca Chang Swee Fun <rebecca.chang@starfivetech.com>
---
 build/config/sysroot.gni | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/build/config/sysroot.gni b/build/config/sysroot.gni
index 18d2d573627c..4f8fa88a50b8 100644
--- a/build/config/sysroot.gni
+++ b/build/config/sysroot.gni
@@ -23,7 +23,8 @@ declare_args() {
   # is empty, default sysroot is calculated.
   use_sysroot = current_cpu == "x86" || current_cpu == "x64" ||
                 current_cpu == "arm" || current_cpu == "arm64" ||
-                current_cpu == "mipsel" || current_cpu == "mips64el"
+                current_cpu == "mipsel" || current_cpu == "mips64el" ||
+                current_cpu == "riscv64"
 }
 
 if (sysroot == "") {
@@ -50,6 +51,8 @@ if (sysroot == "") {
       sysroot = "$target_sysroot_dir/debian_sid_arm-sysroot"
     } else if (current_cpu == "arm64") {
       sysroot = "$target_sysroot_dir/debian_sid_arm64-sysroot"
+    } else if (current_cpu == "riscv64") {
+      sysroot = "$target_sysroot_dir/debian_sid_riscv64-sysroot"
     } else {
       assert(false, "No linux sysroot for cpu: $target_cpu")
     }
-- 
2.25.1

