From b8a12227091a33ee987c8392df8a2f1042bf5d2d Mon Sep 17 00:00:00 2001
From: Rebecca Chang Swee Fun <rebecca.chang@starfivetech.com>
Date: Fri, 15 Jul 2022 07:12:43 +0000
Subject: [PATCH 44/68] BUILD.gn: remove dependency to breakpad

Signed-off-by: Rebecca Chang Swee Fun <rebecca.chang@starfivetech.com>
---
 BUILD.gn | 36 ------------------------------------
 1 file changed, 36 deletions(-)

diff --git a/BUILD.gn b/BUILD.gn
index 6d9fd624b176..847b10c95e9a 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -453,10 +453,6 @@ group("gn_all") {
 
   if (is_linux || is_chromeos || is_android) {
     deps += [
-      "//third_party/breakpad:breakpad_unittests",
-      "//third_party/breakpad:core-2-minidump",
-      "//third_party/breakpad:generate_test_dump",
-      "//third_party/breakpad:minidump-2-core",
       "//tools/dump_process_memory:dump_process",
       "//tools/memory/partition_allocator:all",
     ]
@@ -609,10 +605,6 @@ group("gn_all") {
       "//mojo:mojo_perftests",
       "//services/service_manager/public/cpp",
       "//testing/gmock:gmock_main",
-      "//third_party/breakpad:dump_syms($host_toolchain)",
-      "//third_party/breakpad:microdump_stackwalk($host_toolchain)",
-      "//third_party/breakpad:minidump_dump($host_toolchain)",
-      "//third_party/breakpad:minidump_stackwalk($host_toolchain)",
     ]
 
     if (!is_android) {
@@ -638,9 +630,6 @@ group("gn_all") {
 
   if (is_mac) {
     deps += [
-      "//third_party/breakpad:crash_inspector",
-      "//third_party/breakpad:dump_syms",
-
       # The following are accessibility API tools.
       "//tools/accessibility/inspect:ax_dump_events",
       "//tools/accessibility/inspect:ax_dump_tree",
@@ -684,8 +673,6 @@ group("gn_all") {
         host_os == "win") {
       deps += [ "//chrome/test/mini_installer:mini_installer_tests" ]
     }
-  } else if (!is_android && !is_ios && !is_fuchsia) {
-    deps += [ "//third_party/breakpad:symupload($host_toolchain)" ]
   }
 
   if (is_chromecast) {
@@ -956,7 +943,6 @@ if (is_win) {
       "//net:net_unittests",
       "//printing:printing_unittests",
       "//sql:sql_unittests",
-      "//third_party/breakpad:symupload($host_toolchain)",
       "//ui/base:ui_base_unittests",
       "//ui/gfx:gfx_unittests",
       "//ui/touch_selection:ui_touch_selection_unittests",
@@ -983,7 +969,6 @@ if (is_chromeos_ash) {
       "//ppapi/examples/video_decode",
       "//sandbox/linux:chrome_sandbox",
       "//sandbox/linux:sandbox_linux_unittests",
-      "//third_party/breakpad:minidump_stackwalk($host_toolchain)",
       "//third_party/dawn/src/dawn/tests:dawn_end2end_tests",
       "//third_party/dawn/src/dawn/tests:dawn_unittests",
 
@@ -1147,30 +1132,12 @@ if (!is_ios) {
 
     if (is_android) {
       data_deps += [
-        "//third_party/breakpad:breakpad_unittests",
-        "//third_party/breakpad:dump_syms",
-        "//third_party/breakpad:microdump_stackwalk",
-        "//third_party/breakpad:minidump_dump",
-        "//third_party/breakpad:minidump_stackwalk",
-        "//third_party/breakpad:symupload",
         "//tools/android/forwarder2",
       ]
     } else {
       data_deps += [ "//content/web_test:web_test_common_mojom_js_data_deps" ]
     }
 
-    if (!is_win && !is_android) {
-      data_deps +=
-          [ "//third_party/breakpad:minidump_stackwalk($host_toolchain)" ]
-    }
-
-    if (is_mac) {
-      data_deps += [ "//third_party/breakpad:dump_syms($host_toolchain)" ]
-    }
-
-    if (is_linux || is_chromeos) {
-      data_deps += [ "//third_party/breakpad:dump_syms($host_toolchain)" ]
-    }
 
     if (is_fuchsia) {
       data_deps += [
@@ -1565,9 +1532,6 @@ group("chromium_builder_perf") {
 
     if (is_win) {
       data_deps += [ "//chrome/installer/mini_installer:mini_installer" ]
-    } else {
-      data_deps +=
-          [ "//third_party/breakpad:minidump_stackwalk($host_toolchain)" ]
     }
     if (is_win || is_android) {
       data_deps += [
-- 
2.30.2

