From 703b9ebf14da035efa1ca4bad7b807a407fb81e0 Mon Sep 17 00:00:00 2001
From: Tien Hock Loh <tienhock.loh@starfivetech.com>
Date: Fri, 12 Aug 2022 03:28:05 +0000
Subject: [PATCH 49/68] third_party: breakpad: fix undefined reference

Signed-off-by: Tien Hock Loh <tienhock.loh@starfivetech.com>
---
 third_party/breakpad/BUILD.gn | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/third_party/breakpad/BUILD.gn b/third_party/breakpad/BUILD.gn
index 48defa6ef092..9e1f426e756f 100644
--- a/third_party/breakpad/BUILD.gn
+++ b/third_party/breakpad/BUILD.gn
@@ -617,6 +617,8 @@ if (is_linux || is_chromeos || is_android) {
       "breakpad/src/client/linux/minidump_writer/linux_ptrace_dumper.h",
       "breakpad/src/client/linux/minidump_writer/minidump_writer.cc",
       "breakpad/src/client/linux/minidump_writer/minidump_writer.h",
+      "breakpad/src/client/linux/minidump_writer/pe_file.cc",
+      "breakpad/src/client/linux/minidump_writer/pe_file.h",
       "breakpad/src/client/linux/minidump_writer/proc_cpuinfo_reader.h",
       "breakpad/src/client/minidump_file_writer-inl.h",
       "breakpad/src/client/minidump_file_writer.cc",
-- 
2.30.2

