From d830a2cffd407bc7cefe2c35b8324f3f337526aa Mon Sep 17 00:00:00 2001
From: Tien Hock Loh - swbuildbot <tienhock.loh@starfivetech.com>
Date: Mon, 15 Aug 2022 05:57:59 +0000
Subject: [PATCH 51/68] build: config: compiler: add rpath-link for rv64

Signed-off-by: Tien Hock Loh - swbuildbot <tienhock.loh@starfivetech.com>
---
 build/config/compiler/BUILD.gn | 1 +
 1 file changed, 1 insertion(+)

diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index 4309b0d4e812..c38c4dd1aa64 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -1180,6 +1180,7 @@ config("compiler_cpu_abi") {
       ldflags += [
         "--target=riscv64-linux-gnu",
         "-mno-relax",
+        "-Wl,-rpath-link=./lib",
       ]
     } else if (current_cpu == "s390x") {
       cflags += [ "-m64" ]
-- 
2.30.2

