From 69b5b696114392410d4d08198c1888565a23a974 Mon Sep 17 00:00:00 2001
From: Rebecca Chang Swee Fun <rebecca.chang@starfivetech.com>
Date: Wed, 14 Sep 2022 02:41:26 -0400
Subject: [PATCH 65/68] DEPS: update commit hash for node-nw

Signed-off-by: Rebecca Chang Swee Fun <rebecca.chang@starfivetech.com>
---
 DEPS | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/DEPS b/DEPS
index a75226fbcfb3..57c94e6860f0 100644
--- a/DEPS
+++ b/DEPS
@@ -1806,7 +1806,7 @@ deps = {
     Var('reb_git') + '/v8.git' + '@' +  Var('v8_revision'),
 
   'src/third_party/node-nw':
-    Var('reb_git') + '/node.git' + '@' + '7c46b745ea0ac6cfdac0da84c2e9492c176f8948',
+    Var('reb_git') + '/node.git' + '@' + '90bf0eb747f70320e78a30907e86f34c1c061bc6',
 
   'src/content/nw':
     Var('reb_git') + '/nw.js.git' + '@' + '52deb9b9ed6d4209220ab47975455b73e655b465',
-- 
2.30.2

