diff --git a/BUILD.gn b/BUILD.gn
index f5364875a..fc39965b2 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -217,6 +217,9 @@ config("extra_warnings") {
       "-Wstrict-prototypes",
       "-Wunreachable-code-aggressive",
       "-Wshorten-64-to-32",
+
+      # Workaround for clang unable to locate pci/pci.h header
+      "-isystem../../build/linux/debian_sid_riscv64-sysroot/usr/include/riscv64-linux-gnu",
     ]
   }
 
diff --git a/gni/angle.gni b/gni/angle.gni
index 261cfef0d..3bfd41c72 100644
--- a/gni/angle.gni
+++ b/gni/angle.gni
@@ -78,7 +78,7 @@ declare_args() {
     angle_64bit_current_cpu = true
   } else if (current_cpu == "arm" || current_cpu == "x86" ||
              current_cpu == "mipsel" || current_cpu == "s390" ||
-             current_cpu == "ppc") {
+             current_cpu == "ppc" || current_cpu == "riscv64") {
     angle_64bit_current_cpu = false
   } else {
     assert(false, "Unknown current CPU: $current_cpu")
