diff --git a/gn/standalone/BUILD.gn b/gn/standalone/BUILD.gn
index bd3311460..161d17dee 100644
--- a/gn/standalone/BUILD.gn
+++ b/gn/standalone/BUILD.gn
@@ -243,6 +243,17 @@ config("default") {
         "-msse4.2",
       ]
     }
+  } else if (target_cpu == "riscv64") {
+    cflags += [
+      "-m64",
+      "-mabi=lp64d",
+      "-mno-relax",
+      "--target=riscv64-linux-gnu",
+    ]
+    ldflags += [
+      "-mno-relax",
+      "--target=riscv64-linux-gnu",
+    ]
   }
 
   if (is_linux) {
